WIP: Add support for pgx v5 interface

go-jet opened the pull request:
How to use:
```golang
err = pgxV5.Query(ctx, query, pgxDB, &result)
```
where `pgxDB` can be `*pgx.Tx`, `*pgxpool.Pool` or `*pgx.Conn`.

Default jet generator will work in most cases, except for some postgres specific types like `numeric`, `bit`, `point`, etc...
For those types you'll need to customize generator to use `pgtype.Numeric`, `pgtype.Bits`, `pgtype.Point`, etc...
The full list of such types can be found in this [test](https://github.com/go-jet/jet/blob/48b7a20a31ae89d3dba071955bf129466d89a7db/tests/postgres/pgx_test.go#L168).


The reason why it haven't been merged to master and released is:
- to make it work for uuid types I had to add a hack into hot qrm path. Which I'm not happy about, and this will affect needlessly none pgx users. 
- to get a complete performance improvements of pgx driver probably a bigger refactor of qrm is  required.
- as described above default generator misses a lot of type.

All the tests are passing and you can give it a try.

Even if pgx branch doesn't get into next release I'll keep it in sync with master.

go-jet committed:
Support pgx scan. Initial commit.

go-jet committed:
Merge branch 'master' into pgx

# Conflicts:
#	go.mod
#	go.sum
#	internal/testutils/test_utils.go
#	qrm/scan_context.go
#	tests/postgres/main_test.go
#	tests/postgres/northwind_test.go

go-jet committed:
Quick fix for scanning UUID types using non-standard golang pgx driver.

go-jet committed:
Merge branch 'master' into pgx

DevX86 committed:
feat: add preliminary json support for qrm_pgx.

DevX86 committed:
fix: update queryablePGX interface to be accurate to pgx.

go-jet committed:
Merge pull request #495 from DevX86/pgx

feat: add preliminary json support for qrm_pgx.

go-jet committed:
Add pgxV5 jet utility functions.

go-jet committed:
Update testdata

go-jet committed:
Merge remote-tracking branch 'upstream/master' into pgx

# Conflicts:
#	go.mod
#	go.sum

go-jet committed:
Merge with master.

go-jet committed:
Merge branch 'master' into pgx

# Conflicts:
#	tests/postgres/alltypes_test.go
#	tests/postgres/select_json_test.go

go-jet committed:
Fix test.

DevX86 commented:
Thank you so much for this @go-jet I'm reviewing this and the adapter thread now.

go-jet commented:
> Works great so far! We may have covered this in the past but I'm trying to refresh my memory. Any reason for pgxV5.Query() instead of stmt.QueryPGX() off of the original statement builder I rather have the new one import style in your latest PR than the previous multiline qrm style but figured I'd bring it up. I believe it had something todo with a statement.go needing to be implemented but wanted to get your blessing before trying things.

The reason is because `pgx` is already on version 5 and what guaranties us there want be tomorrow a version 6, 7, 8 etc... I felt this pollutes statements API.

> Doing SELECT_JSON_ARR() with pgxV5.Query() works but for some reason a time.Time on a created_at is giving me the nil timestamp.

Does it work with standard `database/sql` driver? If not, then it is probably something related to a query or a destination.
Could you share your destination?

go-jet committed:
Add pgxV5 Exec method.

gilbsgilbs committed:
fix(qrm): database errors swallowed as ErrNoRows on pgx v5.9.0+

Starting from version 5.9.0, pgx skips the `DescribePortal` round-trip
for cached prepared statements[0]. When a query fails before returning
any data (e.g. unique or FK constraint violation), the server never
sends `RowDescription`, so `rows.FieldDescriptions()` returns nil. Then,
`queryToSlicePgxV5` builds a scan context from "nil" field descriptions,
gets `len == 0`, and returns `(0, nil)` without ever calling
`rows.Err()`, leading callers to see `ErrNoRows` instead of the actual
database error.

This commit fixes this issue by always checking for `rows.Err()` before
returning.

[0] https://github.com/jackc/pgx/blob/master/CHANGELOG.md#590-march-21-2026

go-jet committed:
Merge pull request #567 from gilbsgilbs/pgx-590-fix-wrong-error

[pgx] fix(qrm): database errors swallowed as ErrNoRows on pgx v5.9.0+

go-jet committed:
Merge remote-tracking branch 'upstream/master' into pgx

# Conflicts:
#	go.mod
#	go.sum
#	tests/postgres/main_test.go

go-jet committed:
Fix merge conflicts.

go-jet committed:
Merge remote-tracking branch 'upstream/pgx' into pgx

go-jet committed:
Merge remote-tracking branch 'upstream/master' into pgx
